### PR TITLE
Fix getAnnotationsByDatasetId when no shape is used

### LINKED ISSUE #530: Strange pattern in annotations.py
In the file:

https://github.com/Kitware/UPennContrast/blob/master/devops/girder/annotation_client/annotation_client/annotations.py

There is this code snippet
```
    def getAnnotationsByDatasetId(self, datasetId, shape=None, limit=50, offset=0):
        """
        Get the list of all annotations in the specified dataset

        :param str datasetId: The dataset's id
        :param str shape: optional filter by shape
        :return: A list of annotations
        """
        url = PATHS['annotation_by_dataset'].format(datasetId=datasetId)

        if shape:
            url = '{url}&shape={shape}&limit={limit}&offset{offset}'.format(url=url, shape=shape, limit=limit, offset=offset)

        return self.client.get(url)
```
In that code, it acts a little bit weirdly because if you don't specify shape at all, it gives you 50 annotations of any shape. Ideally, the behavior would be to give you all annotations up to the given limit if you don't specify shape. For instance, right now, this code:

`allAnnotationList = annotationClient.getAnnotationsByDatasetId(datasetId, limit = 1000000)`

only gives you 50 annotations and not the full list.

### PR DESCRIPTION
**Pull request by bruyeret**
It was indeed a bug!
It should be fixed now

Close #530

### CODE REVIEW

File: devops/girder/annotation_client/annotation_client/annotations.py [hunk 1]
```diff
@@ -46,8 +46,9 @@ def getAnnotationsByDatasetId(self, datasetId, shape=None, limit=50, offset=0):
         """
         url = PATHS['annotation_by_dataset'].format(datasetId=datasetId)
 
+        url = f'{url}&limit={limit}&offset{offset}'
         if shape:
-            url = '{url}&shape={shape}&limit={limit}&offset{offset}'.format(url=url, shape=shape, limit=limit, offset=offset)
+            url = f'{url}&shape={shape}'
```
Comment (luciemac) [inline devops/girder/annotation_client/annotation_client/annotations.py:51]: I think you should add a else case here

Comment (bruyeret) [reply devops/girder/annotation_client/annotation_client/annotations.py:51]: What should I put in the else case?

Comment (arjunrajlab) [reply devops/girder/annotation_client/annotation_client/annotations.py:51]: I think that if shape is not given, it should just return all annotations (subject to limit and offset).
Also, while we are at it, is there a way to add a filter by tag? Like, to get all annotations by tag? That would be useful.
Also, let's change default limit to 1,000,000. I think that is a reasonable number. 50 is pretty small.

Comment (bruyeret) [reply devops/girder/annotation_client/annotation_client/annotations.py:51]: Returning all annotations and taking limit and offset into account is now the default behavior when no shape is given
I can add the feature to filter by tags and increase the default limit
In the tag filtering, what would be the algorithm?
- Get all annotations with exactly the given tags
- Get all annotations with all the given tags and eventually other tags
- Get all annotations with at least one of the given tags

**Comment by luciemac** [comment]
LGTM

File: devops/girder/annotation_client/annotation_client/annotations.py (hunk 1 above)
Comment (arjunrajlab) [reply devops/girder/annotation_client/annotation_client/annotations.py:51]: Yeah, I wasn't sure exactly what to do there :). I think it should be all of the given tags and potentially other additional tags.

**Comment by bruyeret** [comment]
I added the tags filter to the endpoint and propagated the changes to the annotation client
What do you think @arjunrajlab?

**Comment by arjunrajlab** [comment]
I will try and test it later today or tomorrow!

**Comment by bruyeret** [comment]
I forgot to mention that girder has to be rebuilt

**Review by arjunrajlab — APPROVED** [review]
Looks good to me, ready to merge!

**Pull request merged**